esx_service/vmdk_ops.py

```diff
@@ -668,20 +670,31 @@ def vm_uuid2name(vm_uuid):
         return None
     return vm.config.name
 
-# Return error, or None for OK.
-def attachVMDK(vmdk_path, vm_uuid):
-    vm = findVmByUuid(vm_uuid)
-    logging.info("*** attachVMDK: %s to %s VM uuid = %s",
-                 vmdk_path, vm.config.name, vm_uuid)
-    return disk_attach(vmdk_path, vm)
 
+def attach_or_detach_VMDK(cmd, vmdk_path, vm_name, bios_uuid, vc_uuid):
+    # note: vc_uuid is the last one to avoid reworkign tests which use positional args and
+    # not aware of vc_uuid
+    """Attaches (if cmd="attach" or detaches a  VMDK.
+    Returns json reply to pass upstairs"""
+
+    logging.info("*** %s VMDK %s to VM '%s' , bios uuid = %s, VC uuid=%s)",
+                 cmd, vmdk_path, vm_name, bios_uuid, vc_uuid)
+    vm = None
+    if vc_uuid:
+        vm = findVmByUuid(vc_uuid)
+    if not vm: # either vc_uuid is not even passed, or we failed to find the VM by VC uuid:
+        logging.warning("Failed to find VM by VC UUID %s, trying BIOS UUID %s", vc_uuid, bios_uuid)
+        vm = findVmByUuid(bios_uuid)
+    if not vm: # can find VM by VC or BIOS uuid
+        msg = "Failed to find VM object for %s (bios %s vc %s)" % (vm_name, bios_uuid, vc_uuid)
+        logging.error(msg)
+        return err(msg)
 
-# Return error, or None for OK.
-def detachVMDK(vmdk_path, vm_uuid):
-    vm = findVmByUuid(vm_uuid)
-    logging.info("*** detachVMDK: %s from %s VM uuid = %s",
-                 vmdk_path, vm.config.name, vm_uuid)
-    return disk_detach(vmdk_path, vm)
+    if vm.config.name != vm_name:
```

Comment:
Can it happen that vm object is not null but vm.config is / or vm.config.name is.
We can have a check or a simple try catch wrap.